### ISSUE TITLE
Access to older development builds

### ISSUE THREAD

**Issue by wcislo-saleor** [HaxeFoundation/haxe#10988]
Is there any build archive that stores development builds other than https://build.haxe.org/builds/haxe/?

I'm currently trying to work with a certain project that seems to be requiring specific haxe build (from commit https://github.com/HaxeFoundation/haxe/commit/1ec0856). This development build doesn't seem to be available in the build archive anymore.

**Comment by Simn** [comment]
I don't think there's anything official other than that. Why exactly are you stuck on a random dev version?

**Comment by wcislo-saleor** [comment]
Thanks for the prompt answer.

There is this abandoned project (https://github.com/BeeInventor/keycloak-cognito-migration/blob/main/.haxerc) that relies on this haxe build. Once I tried compiling it with haxe `4.2.5`, `4.2.4` and around dozen development builds starting from the oldest one available I either experienced complication errors or runtime errors.

I'm not entirely sure it's due to haxe thought, I wanted to check that by using this exact haxe build.

**Comment by Simn** [comment]
Could you post the compilation errors you are getting with the latest development version?

**Comment by wcislo-saleor** [comment]
We've found an alternative way around our problem that does not require use of keycloak-cognito-migration project.

----

Compilation of https://github.com/BeeInventor/keycloak-cognito-migration/commit/8f6ab761cdf94f1e7ca0b017d459deefba390b7a with the oldest available development build `b687030` as well as with the newest development build `85816bd` results in following error.

```[INFO] --- haxe-maven-plugin:1.0.0:build (default) @ keycloak-cognito-migration ---
/home/keycloak/haxe/haxe_libraries/tink_core/2.0.2/github/8fb0b9aa4de933614b5a04cc88da871b89cb8c6a/src/tink/core/Promise.hx:318: characters 5-21 : Recursive implicit cast
/home/keycloak/haxe/haxe_libraries/tink_core/2.0.2/github/8fb0b9aa4de933614b5a04cc88da871b89cb8c6a/src/tink/core/Promise.hx:315: characters 5-21 : Recursive implicit cast
/home/keycloak/haxe/haxe_libraries/tink_core/2.0.2/github/8fb0b9aa4de933614b5a04cc88da871b89cb8c6a/src/tink/core/Promise.hx:312: characters 5-21 : Recursive implicit cast
/home/keycloak/haxe/haxe_libraries/tink_core/2.0.2/github/8fb0b9aa4de933614b5a04cc88da871b89cb8c6a/src/tink/core/Promise.hx:343: characters 5-33 : Recursive implicit cast
/home/keycloak/haxe/haxe_libraries/tink_core/2.0.2/github/8fb0b9aa4de933614b5a04cc88da871b89cb8c6a/src/tink/core/Promise.hx:340: characters 5-33 : Recursive implicit cast
/home/keycloak/haxe/haxe_libraries/tink_core/2.0.2/github/8fb0b9aa4de933614b5a04cc88da871b89cb8c6a/src/tink/core/Promise.hx:337: characters 5-33 : Recursive implicit cast
src/haxe/com/beeinventor/keycloak/CognitoMigrationUserStorageProviderFactory.hx:217: lines 217-232 : Unknown<0> cannot be constructed
src/haxe/com/beeinventor/keycloak/CognitoMigrationUserStorageProviderFactory.hx:217: lines 217-232 : ... For function argument 'task'
java.io.IOException: Stream closed
        at java.base/java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:176)
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:342)
        at java.base/java.io.InputStream.readNBytes(InputStream.java:396)
        at java.base/java.io.InputStream.readAllBytes(InputStream.java:333)
        at com.kevinresol.HaxeBuildMojo.execute(HaxeBuildMojo.java:87)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:972)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
```

Compilation of https://github.com/BeeInventor/keycloak-cognito-migration/commit/8f6ab761cdf94f1e7ca0b017d459deefba390b7a with `4.2.5` build results in successful compliation but the plugin fails when being loaded by Keycloak (runtime). I'm unable to provide Keycloak logs.

**Comment by kLabz** [comment]
Did you try with more recent tink libs and latest haxe nightlies?
